### ISSUE TITLE
Add support for volumes on creation for scaleway

### ISSUE THREAD

**Issue by dcharbonnier** [hashicorp/terraform#9499]
The faster way to create the volumes is during the creation of the server resource, no up and down of the server, no volume migration (very slow operation)
My use case of volumes is to have more space available on a server than the regular 50Go, not to move volumes from one container to an other one. I suspect that it's the use case for most of the people and I would say more over if they use terraform.

Scaleway API provide an helper for server creation : [CreateServer(api *ScalewayAPI, c *ConfigCreateServer) (string, error)](https://github.com/scaleway/scaleway-cli/blob/master/pkg/api/helpers.go#L293)
We don't import the helper in terraform for good reason but this logic seams good and more important validated by @scaleway

The volumes created during this step should not be considered as "resources" (same as the root volume) and should be destroyed when the server is terminated.

```
resource "scaleway_server" "test" {
  name = "test"
  image = "${data.scaleway_image.docker.id}"
  type = "${var.type}"
  additional_volume {
    size_in_gb = 50
    type = "l_ssd"
  }
  additional_volume {
    size_in_gb = 50
    type = "l_ssd"
  }
}
```

There is already an implementation https://github.com/hashicorp/terraform/pull/8678 but this PR address different problems and some have been better solved later, IMO this PR should be rejected.

related issues #9417, #9254, #8678, #9493

**Comment by dcharbonnier** [comment]
ping @nicolai86 and @tboerger

**Comment by nicolai86** [comment]
I think we should settle for a more verbose version of the additional volumes, like this: 

```
resource "scaleway_server" "test" {
  name = "test"
  image = "${data.scaleway_image.docker.id}"
  type = "${var.type}"
  additional_volume {
    size_in_gb = 50
    type = "l_ssd"
  }
  additional_volume {
    size_in_gb = 50
    type = "l_ssd"
  }
}
```

this way we're sure to support different storage types if Scaleway decides to include them in the future.

related: #8678, but the approach is the less verbose one which I believe will lead to problems down the road.

**Comment by tboerger** [comment]
I personally don't need this approach since I'm only adding volumes as storage, and even if the server gets destroyed I don't want to delete that storage in every case :)

**Comment by nicolai86** [comment]
@tboerger it's a necessity to be able to use VC1M and VC1L instance types.

**Comment by tboerger** [comment]
> @tboerger it's a necessity to be able to use VC1M and VC1L instance types.

This can also be handled differently like https://github.com/hashicorp/terraform/pull/8678/files#diff-da07af34fc933e15f9fdcd3e6b3ee3baR74

**Comment by dcharbonnier** [comment]
@tboerger applying the resources is not incompatible obviously it's required and the PR that is about to be merged fix the issue related to this.

@nicolai86 ok for the syntax, that's not very easy to add x volume with x as a var but with a template it's doable, I update the issue

**Comment by dcharbonnier** [comment]
@stack72 this issue is not resolved, there is work done around it and that's why we reference this issue but its not implemented for now and still required.

**Comment by nicolai86** [comment]
@dcharbonnier correct me if I'm wrong but I was under the impression that the recently merged #9695 resolves this very issue, no?

**Comment by dcharbonnier** [comment]
Sorry, you've been so fast on developing it and merging that i couldn't believe it.
